discovery-infra/start_discovery.py

```diff
@@ -357,13 +447,33 @@ def nodes_flow(client, cluster_name, cluster, image_path):
             validate_dns(client, cluster.id)
 
 
-def main():
-    client = None
-    cluster = {}
+def execute_day2_flow(image_folder, internal_cluster_name, num_workers, api_vip_ip, api_vip_dnsname):
+    utils.recreate_folder(image_folder)
+    client = assisted_service_api.create_client(
+            url=utils.get_assisted_service_url_by_args(args=args)
+            )
+    cluster_id = str(uuid.uuid4())
+    random_postfix = cluster_id[:8]
+    ui_cluster_name = internal_cluster_name + f'-{random_postfix}'
+    cluster = client.create_day2_cluster(
+            ui_cluster_name, cluster_id, **_day2_cluster_create_params()
+            )
 
-    internal_cluster_name = f'{args.cluster_name or consts.CLUSTER_PREFIX}-{args.namespace}'
-    log.info('Cluster name: %s', internal_cluster_name)
+    set_cluster_pull_secret(client, cluster_id, args.pull_secret)
```

Comment:
can't we pass pull secret on create?

Reply:
no, we day2 register API does not require secret, user will pass it as part of create/download ISO